service/rtc/sfu.go

```diff
@@ -484,7 +488,21 @@ func (s *Server) InitSession(cfg SessionConfig, closeCb func() error) error {
 
 			s.log.Debug("received screen sharing stream", mlog.String("streamID", streamID), mlog.String("sessionID", us.cfg.SessionID))
 
-			outScreenTrack, err := webrtc.NewTrackLocalStaticRTP(params.RTPCodecCapability, genTrackID(trackTypeScreen, us.cfg.SessionID), random.NewID(), webrtc.WithRTPStreamID(remoteTrack.RID()))
+			// To improve concurrency and support larger calls we create NumCPU output tracks and randomly distribute the receivers among them.
+			createOutScreenTracks := func(num int) ([]*webrtc.TrackLocalStaticRTP, error) {
+				outTracks := make([]*webrtc.TrackLocalStaticRTP, num)
+				for i := 0; i < num; i++ {
+					outTrack, err := webrtc.NewTrackLocalStaticRTP(params.RTPCodecCapability,
+						genTrackID(trackTypeScreen, us.cfg.SessionID), random.NewID(), webrtc.WithRTPStreamID(remoteTrack.RID()))
+					if err != nil {
+						return nil, fmt.Errorf("failed to create screen track")
+					}
+					outTracks[i] = outTrack
+				}
+				return outTracks, nil
+			}
+
+			outScreenTracks, err := createOutScreenTracks(runtime.NumCPU())
```

Comment:
I am still debating whether `runtime.NumCPU()` is the optimal value. Using double that has shown even better CPU utilization but it comes with tradeoffs as one huge call could potentially bring the whole service to its knees, including other calls. So it may be better to be conservative to support a multi-call scenario.